src/compiler.ts

```diff
@@ -6191,8 +6191,8 @@ export class Compiler extends DiagnosticEmitter {
       if (flow.isAnyLocalFlag(localIndex, LocalFlags.ANY_RETAINED)) {
         valueExpr = this.makeReplace(
           valueExpr,
-          module.local_get(localIndex, type.toNativeType()),
           type,
+          module.local_get(localIndex, type.toNativeType()),
```

Comment:
Something odd about the order here now, Shouldn't this be `valueExpr, valueType, EXPR, type`?